.gitlab-ci.yml

```diff
@@ -11,6 +15,59 @@ include:
   - project: 'Northern.tech/Mender/mendertesting'
     file: '.gitlab-ci-github-status-updates.yml'
 
+.set_eks_helmci_vars: &set_eks_helmci_vars
+  - export AWS_ACCESS_KEY_ID=$CI_EKSCTL_ACCESS_KEY_ID_TEST
+  - export AWS_SECRET_ACCESS_KEY=$CI_EKSCTL_AWS_SECRET_ACCESS_KEY_TEST
+  - export AWS_DEFAULT_REGION=eu-central-1
+
+build:pipeline-toolbox-image:
+  tags:
+    - mender-qa-worker-generic-light
+  stage: .pre
+  image: docker
+  services:
+    - docker:dind
+  rules:
+    - if: '$CI_COMMIT_BRANCH == "master" || $CI_COMMIT_BRANCH == "master-next" || $RUN_HELM_CHART_INSTALL == "true" || $CI_COMMIT_TAG'
+  script:
+    - echo $CI_REGISTRY_PASSWORD | docker login -u $CI_REGISTRY_USER $CI_REGISTRY --password-stdin
+    - |
+      docker build \
+            --build-arg BUILDKIT_INLINE_CACHE=1 \
+            --cache-from ${CI_REGISTRY_IMAGE}:${PIPELINE_TOOLBOX_TAG} \
+            -t ${CI_REGISTRY_IMAGE}:${PIPELINE_TOOLBOX_TAG} \
+            --build-arg KUBECTL_VERSION=${CI_KUBECTL_VERSION} \
+            --build-arg HELM_VERSION=${CI_HELM_VERSION} \
+            --build-arg KUBECONFORM_VERSION=${CI_KUBECONFORM_VERSION} \
+            -f tests/Dockerfile \
+            tests/
+    - docker push ${CI_REGISTRY_IMAGE}:${PIPELINE_TOOLBOX_TAG}
+
+build:setup_eks_cluster:
+  image: ${CI_REGISTRY_IMAGE}:${PIPELINE_TOOLBOX_TAG}
+  stage: .pre
+  needs: ["build:pipeline-toolbox-image"]
+  rules:
+    - if: '$CI_COMMIT_BRANCH == "master" || $CI_COMMIT_BRANCH == "master-next" || $RUN_HELM_CHART_INSTALL == "true" || $CI_COMMIT_TAG'
+  before_script:
+  script:
+    - *set_eks_helmci_vars
+    - |
+      echo "INFO - Temporary EKS cluster setup"
+      eksctl create cluster \
+        --name ${EKS_CLUSTER_NAME} \
+        --tags "env=helm-ci-test" \
+        --asg-access \
+        --spot \
+        --instance-types m6a.xlarge,t3.xlarge \
+        --nodes 4
+    - echo "INFO - assigning SSO roles for debugging failed tests:"
+    - eksctl create iamidentitymapping --cluster ${EKS_CLUSTER_NAME} --arn arn:aws:iam::017659451055:role/AWSReservedSSO_AdministratorAccess_d80c0803237d713a --username "admin:{{SessionName}}" --group "system:masters" --no-duplicate-arns
+    - eksctl create iamidentitymapping --cluster ${EKS_CLUSTER_NAME} --arn arn:aws:iam::017659451055:role/AWSReservedSSO_PowerUserAccess_28a0f819b31196fb --username "admin:{{SessionName}}" --group "system:masters" --no-duplicate-arns
+    - eksctl create iamidentitymapping --cluster ${EKS_CLUSTER_NAME} --arn arn:aws:iam::017659451055:role/AWSReservedSSO_EKSAccess_8c2574e21d9fc21c --username "admin:{{SessionName}}" --group "system:masters" --no-duplicate-arns
+    - echo "DEBUG - kubectl cluster version:"
+    - kubectl version
```

Comment:
Isn't this a bit overkill? Why not reuse an existing cluster and use a dedicated namespace for the tests?

Reply:
Yep, it's way overkill. But having this on the existing cluster won't work for the long term, when we'll put the staging behind the VPN: so we can push helm install no more...

Also consider other pros:
* you always test the helm chart over the latest EKS cluster
* you don't need to have staging tuned also for the pipelines (e.g.: the node affinities rules)
* here you can also add more clusters: GKE, AKS, if needed in the future.